src/states/states.ts

```diff
@@ -355,3 +355,47 @@ export class TerminalState extends BaseState {
     this.type = 'Terminal';
   }
 }
+
+export type AnyState =
+  | DeathState
+  | SymptomState
+  | DiagnosticReportState
+  | MultiObservationState
+  | ObservationState
+  | VitalSignState
+  | SupplyListState
+  | BaseState
+  | EncounterEndState
+  | EncounterState
+  | ConditionEndState
+  | ConditionOnsetState
+  | AllergyEndState
+  | AllergyOnsetState
+  | MedicationEndState
+  | MedicationOrderState
+  | CarePlanEndState
+  | CarePlanStartState
+  | ProcedureState
+  | ImagingStudyState
+  | DeviceEndState
+  | DeviceState
+  | TerminalState
+  | InitialState;
+
+export type HasCodes =
+  | EncounterState
+  | ConditionOnsetState
+  | AllergyOnsetState
+  | MedicationOrderState
+  | CarePlanStartState
+  | ProcedureState
+  | ObservationState
+  | MultiObservationState
+  | DiagnosticReportState;
+
+export const doesHaveCodes = (tbd: AnyState): tbd is HasCodes => {
```

Comment:
Couldn't this be added to the `BaseState` class? Is the intention to return true if the specific state type has a `codes` field or is it only if the field is actually defined with codes?